src/middlewared/middlewared/scripts/ctdb_events.py

```diff
@@ -0,0 +1,276 @@
+            return
+
+        if not os.path.exists(CTDB_SERVICE_FILE):
+            return
+
+        self.load_service_file()
+        for srv in self.cl_services.values():
+            if srv['enable']:
+                self.client.call('service.restart', srv['name'])
+
+    def shutdown(self):
+        """
+        This shuts down all services managed by ctdb and is
+        triggered by ctdb shutting down.
+        """
+        return
+
+    def monitor(self):
+        if not os.path.exists(CTDB_SERVICE_FILE):
+            return
+
+        if not self.client.call('system.ready'):
+            return
+
+        payload = {
+            'event': 'MONITOR',
+            'status': 'SUCCESS',
+        }
+
+        last_failed = os.path.exists(CTDB_MONITOR_FAILED_SENTINEL)
+
+        try:
+            self.check_ctdb_shared_volume()
+        except Exception as e:
+            payload = {
+                'event': 'MONITOR',
+                'status': 'FAILURE',
+                'reason': str(e),
+                'service': 'ctdb_shared_volume',
+            }
+
+        if not payload['status'] == 'FAILURE':
+            self.load_service_file()
+            for srv in self.cl_services.values():
+                srvinfo = self.client.call('service.query', [['service', '=', srv['name']]], {'get': True})
+                final_state = 'STOPPED'
+                if srv['enable']:
+                    if srvinfo['state'] != 'RUNNING':
+                        self.logger.warning('%s: managed service is not running. Attempting to start.',
+                                            srv['name'])
+                        try:
+                            self.client.call('service.start', srv['name'], {'silent': False})
+                            final_state = 'RUNNING'
+                        except Exception as e:
+                            self.logger.warning('%s: failed to start managed service.',
+                                                srv['name'], exc_info=True)
+
+                            errmsg = f'{srv["name"]}: service failed to start: {e}'
+                            payload = {
+                                'event': 'MONITOR',
+                                'status': 'FAILURE',
+                                'reason': errmsg,
+                                'service': srv['name'],
+                            }
+                            break
+
+                    if not srvinfo['enable']:
+                        self.client.call('service.update', srvinfo['id'], {'enable': True})
+
+                else:
+                    if srvinfo['enable']:
+                        self.client.call('service.update', srvinfo['id'], {'enable': False})
+
+                    if srvinfo['state'] == 'RUNNING':
+                        self.client.call('service.stop', srv['name'])
+
+                self.node_status[srv['name']] = final_state
+
+        if last_failed and payload['status'] == 'SUCCESS':
```

Comment:
Do we care if the sentinel exists if the `payload['status']` is success? Would it not be better to ju;st check for `SUCCESS` and remove the sentinel always suppressing filenotfounderror?

Reply:
There was another indentation error. Idea is to use the sentinel file to determine whether our service state has changed. Although when I was originally writing this I didn't keep track of run state of services in the gluster volume so maybe I can use that instead of sentinel.

Reply:
Service state change is deciding factor for whether we passing monitoring results to middleware.